reconcile/utils/terrascript_aws_client.py

```diff
@@ -2255,7 +2277,54 @@ def populate_tf_resource_sqs(self, spec):
             tf_resources.append(tf_resource)
 
         self.add_resources(account, tf_resources)
+    
+    def populate_tf_resource_sns(self, spec):
+        account = spec.provisioner_name
+        identifier = spec.identifier
+        common_values = self.init_values(spec)
+        output_prefix = spec.output_prefix
+        policy = common_values.get("inline_policy")
+        region = common_values.get("region") or self.default_regions.get(account)
+        
+        values = {}
+        if 'fifo_topic' in common_values.keys():
+            fifo_topic = common_values.get("fifo_topic")
+        else:
+            fifo_topic = False
+        if fifo_topic:
+            topic_name = identifier+('.fifo')
+        else:
+            topic_name = identifier
+        
+        values["name"] = topic_name 
+        values["policy"] = policy 
+        values["fifo_topic"] = fifo_topic 
+
+        tf_resources = []
+        self.init_common_outputs(tf_resources, spec)
+        tf_resource = aws_sns_topic(identifier,**values)
+        tf_resources.append(tf_resource)
+        
+        if "subscriptions" in common_values.keys():
+            subscriptions = common_values.get("subscriptions")
+            counter = 0
+            for sub in subscriptions:
+                counter += 1
+                sub_values = {}
+                sub_values["topic_arn"] = "${aws_sns_topic"+ "." + identifier + ".arn}"
+                sub_values["protocol"] = sub["protocol"]
+                sub_values["endpoint"] = sub["endpoint"]
+                sub_identifier = f"{identifier}_aws_sns_topic_subscription_{counter}"
```

Comment:
Much like the other case, would these be recreated if a subscription was removed?

Reply:
I would think so. But let me test it out.

Reply:
Similar to `event_notifications`.  The primary key should be combination of topic name and `protocol`.